playbook/app/pb_kits/playbook/pb_timestamp/_timestamp.jsx

```diff
@@ -2,36 +2,116 @@
+    showUser = false,
+    showTimezone = false,
+    variant = 'default',
+  } = props
+
+  const ariaProps = buildAriaProps(aria)
   const dataProps = buildDataProps(data)
-  const pbCss = buildCss('pb_timestamp_kit')
+  const classes = classnames(
+    buildCss('pb_timestamp_kit', align, {
+      dark: dark,
+      variant: variant,
+    }),
+    globalProps(props),
+    className
+  )
+
+  const currentYear = new Date().getFullYear().toString()
+  const dateTimestamp = new DateTime({ value: timestamp, zone: timezone })
+  const dateDisplay = `${dateTimestamp.toMonth()} ${dateTimestamp.toDay()}`
+  const timeDisplay = `${dateTimestamp.toHour()}:${dateTimestamp.toMinute()}${dateTimestamp.toMeridian()}`
+
+  const fullTimeDisplay = (dateTimestamp, timeDisplay, timezone, showTimezone) => {
+    if (showTimezone == true && timezone.length > 0) {
+      timeDisplay = timeDisplay + ' ' + dateTimestamp.toTimezone()
+    }
+    return timeDisplay
+  }
+
+  const fullDateDisplay = (dateTimestamp, currentYear, timezone, showTimezone) => {
+    let fullDisplay = `${dateTimestamp.toMonth()} ${dateTimestamp.toDay()}`
+    if (dateTimestamp.toYear() > currentYear) {
+      fullDisplay = `${fullDisplay}, ${dateTimestamp.toYear()}`
+    }
+    return `${fullDisplay} ${' \u00b7 '} ${fullTimeDisplay(dateTimestamp, timeDisplay, timezone, showTimezone)}`
+  }
+
+  const formattedUpdatedString = (showUser, text, dateTimestamp, timeDisplay) => {
+    switch (variant) {
+    case 'elapsed': {
+      const elapsedUserDisplay = (showUser == true && text.length > 0) ? ` by ${text}` : ''
+      return `Last updated ${elapsedUserDisplay} ${dateTimestamp.value.fromNow()}`
+    }
+    case 'updated': {
+      const updatedUserDisplay = (showUser == true && text.length > 0) ? ` by ${text}` : ''
+      return `Last updated ${updatedUserDisplay} on ${dateDisplay} at ${timeDisplay}`
+    }
+    }
+  }
 
   return (
     <div
+        {...ariaProps}
         {...dataProps}
-        className={classnames(pbCss, globalProps(props), className)}
-        id={id}
+        className={classes}
     >
-      <Caption
-          size="xs"
-          tag="span"
-          text={text}
-      />
+      <div className="pb_timestamp_kit">
+        <If condition={variant == 'updated' || variant == 'elapsed'}>
+          <Caption
+              dark={dark}
+              size="xs"
+              text={formattedUpdatedString(showUser, text, dateTimestamp, timeDisplay)}
+          />
+          <Else />
+          <If condition={showDate}>
+            <Caption
+                dark={dark}
+                size="xs"
+                text={timestamp ? fullDateDisplay(dateTimestamp, currentYear, dateDisplay, timezone, showTimezone) : text}
+            />
+            <Else />
+            <Caption
+                dark={dark}
+                size="xs"
+                text={fullTimeDisplay(dateTimestamp, timeDisplay, timezone, showTimezone)}
+            />
+          </If>
+        </If>
```

Comment:
Similar to the above comment on the Rails `.erb` template, you can apply the same thinking here and abstract in a very similar way in order to dry this up a bit.

Reply:
Agreed!